olp-cpp-sdk-core/src/client/OlpClient.cpp

```diff
@@ -400,7 +414,12 @@ HttpResponse OlpClient::CallApi(
   auto response =
       SendRequest(network_request, settings_, retry_settings, context);
 
-  for (int i = 0; i < retry_settings.max_attempts && !context.IsCancelled();
+  // Make sure that we don't wait longer than `timeout` in retry settings
```

Comment:
```suggestion
```
Redundant comment.

Reply:
I would argue about removal of this comment - I think it would be easier for reader to understand why we cap the max wait time.